### DISCUSSION TITLE
Running Offline installation

### DISCUSSION THREAD

**Discussion by ANISHSAJIKUMAR** [autobase-tech/autobase#210] [Q&A]
Hii I am using centos 8 . It really hard to run ansible script in offline environment is there any was you run this script in offline environment without internet . Once I used the same script in redhat offline environment and struggled a lot and commented lot of thing and some how it worked. But it's really hard . So is there any way to run it in a machine without internet. Like back servers ?

**Comment by firedragon1998** [comment]
i am also looking for the same

**Reply by vitabaks** [reply]
it's really hard

**Reply by vitabaks** [reply]
If you want to use the installation method "file".
You need to independently determine all the necessary dependencies of rpm packages for your version of the Linux distribution.

**Reply by vitabaks** [reply]
I usually use a [proxy](https://github.com/vitabaks/postgresql_cluster/blob/master/vars/main.yml#L6) server in an environment where the server does not have direct internet access.

**Reply by ANISHSAJIKUMAR** [reply]
In this server we don't have internet access and there is no proxy server . So the only way is to download rpm and install . I am downloading rpms but lot of dependencies the server even don't have normal dependencies available.

**Reply by ANISHSAJIKUMAR** [reply]
Currently I am using vpn to connect to the remote windows machine and then ssh in to 3 completely offline server . So in this how should I use proxy repo

**Reply by vitabaks** [reply]
In order for this to work, you need to define all the dependencies. 

to make this task a little easier, I recommend trying to use the rpm package for patroni instead of pip

- `installation_method: file`
- `patroni_installation_method: rpm`
- `patroni_rpm_package_file`:


https://download.postgresql.org/pub/repos/yum/common/redhat/rhel-8-x86_64/patroni-2.1.4-1.rhel8.x86_64.rpm

**Reply by ANISHSAJIKUMAR** [reply]
Can we do everything manually like installation of only patroni etcd and Postgres 14 in centos. Is it hard ? Only want Postgres ha is it doable or really hard ?

**Reply by vitabaks** [reply]
offline installation is not easy but possible.

Unfortunately, I can't allocate enough time for this project to help with offline installation. At least until there is sufficient [sponsorship](https://github.com/vitabaks/postgresql_cluster#sponsor-this-project) of this project.
